actor/src/main/scala/org/apache/pekko/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -41,10 +43,17 @@ import pekko.util.PrettyDuration._
 private[io] final class AsyncDnsResolver(
     settings: DnsSettings,
     cache: SimpleDnsCache,
-    clientFactory: (ActorRefFactory, List[InetSocketAddress]) => List[ActorRef])
+    clientFactory: (ActorRefFactory, List[InetSocketAddress]) => List[ActorRef],
+    idGenerator: IdGenerator)
     extends Actor
     with ActorLogging {
 
+  def this(
```

Comment:
This maintains binary compatibility a new parameter with a default value does not.

Reply:
Seems useful to maintain compatibility but we can document any places where we need to break compatibility. For me, Pekko is a new project and users already have to migrate package names, config names, some class names - so we are already not fully compatible with Akka.